setup.py

```diff
@@ -1,38 +1,32 @@
-import os
 import re
-
+from pathlib import Path
 from setuptools import find_packages, setup
 
-# NOTE: If updating requirements make sure to also check Pipfile for any locks
-# NOTE: When updating botocore make sure to update awscli/boto3 versions below
-install_requires = [
-    # pegged to also match items in `extras_require`
-    'botocore>=1.33.2,<1.33.14',
-    'aiohttp>=3.7.4.post0,<4.0.0',
-    'wrapt>=1.10.10, <2.0.0',
-    'aioitertools>=0.5.1,<1.0.0',
-]
+_this_dir = Path(__file__).parent
 
+# NOTE: When updating botocore make sure to update install-requires.txt
 extras_require = {
     'awscli': ['awscli>=1.31.2,<1.31.14'],
     'boto3': ['boto3>=1.33.2,<1.33.14'],
 }
 
 
-def read(f):
-    return open(os.path.join(os.path.dirname(__file__), f)).read().strip()
+def read(file_name: str):
+    return (_this_dir / file_name).read_text().strip()
```

Comment:
`pathlib` sure is more idiomatic. And I love type hints :-) Why not annotate the return type (`-> str`), while we're at it?

Reply:
reverted file